### PR TITLE
Add support for NCT6683D series

### PR DESCRIPTION
**Pull request by chencaidy**
Add support for Nuvoton NCT6683D/NCT6686D/NCT6687D, 
it tested on ThinkCentre M920x with NCT6686D EC, use Macmini8,1 SMBIOS

I tested unplug the fan connector, F0Ac return zero soon.
smcread logs:
[F0Ac] type [fpe2] 66706532 len [ 2] attr [80] -> 193C
[F1Ac] type [fpe2] 66706532 len [ 2] attr [80] -> 0000
[F2Ac] type [fpe2] 66706532 len [ 2] attr [80] -> 0000
[F3Ac] type [fpe2] 66706532 len [ 2] attr [80] -> 0000
[F4Ac] type [fpe2] 66706532 len [ 2] attr [80] -> 0000

But voltage function seems are programable regs and programed by OEM, 
so it marked with todo.

### CODE REVIEW

**Comment by chencaidy** [comment]
Hi @vit9696,

Voltage read seems working on NCT6683,

It reported VIN0, VIN1 and VIN3 from ThineCentre M920x.

```
    | |   | +-o SMCSuperIO  <class SMCSuperIO, id 0x1000002c9, !registered, !matched, active, busy 0, retain 6>
    | |   | |   {
    | |   | |     "IOClass" = "SMCSuperIO"
    | |   | |     "CFBundleIdentifier" = "ru.joedm.SMCSuperIO"
    | |   | |     "IOProviderClass" = "IOPCIDevice"
    | |   | |     "VersionInfo" = "REL-123-2021-05-07"
    | |   | |     "IOPCIClassMatch" = "0x06010000&0xffff0000"
    | |   | |     "IOPowerManagement" = {"CapabilityFlags"=32770,"MaxPowerState"=1,"CurrentPowerState"=1}
    | |   | |     "ChipName" = "Nuvoton NCT6686D"
    | |   | |     "IOResourceMatch" = "ACPI"
    | |   | |     "IOProbeScore" = 0
    | |   | |     "Sensors" = {"VIN12"=<00000000>,"VTT"=<00000000>,"3VCC"=<00000000>,"VIN11"=<00000000>,"VIN3"=<1b2fdd3e>,"VIN16"=<00000000>,"VIN8"=<00000000>,"VIN10"=<00000000>,"SYSFAN"=0,"VBAT"=<00000000>,"VIN1"=<b91e853f>,"CPUFAN"=1602,"VIN6"=<00000000>,"VIN15"=<00000000>,"VREF"=<00000000>,"VIN14"=<00000000>,"VIN4"=<00000000>,"AUXFAN0"=0,"3VSB"=<00000000>,"VIN9"=<00000000>,"VIN13"=<00000000>,"VIN2"=<00000000>,"AUXFAN1"=0,"VIN7"=<00000000>,"VIN0"=<f0a7c63f>,"AUXFAN2"=0,"AVSB"=<00000000>,"VIN5"=<00000000>}
    | |   | |     "IOMatchCategory" = "SMCSuperIO"
    | |   | |     "CFBundleIdentifierKernel" = "ru.joedm.SMCSuperIO"
    | |   | |   }
```

File: Sensors/SMCSuperIO/NuvotonDevice.hpp [hunk 1]
```diff
@@ -29,6 +29,19 @@ namespace Nuvoton {
...
+	static constexpr uint8_t NUVOTON_6683_DATA_REGISTER_OFFSET	= 0x06;
+	static constexpr uint8_t NUVOTON_6683_EVENT_REGISTER_OFFSET	= 0x07;
+	static constexpr uint8_t NUVOTON_6683_MON_NUMS				= 32;
+	static constexpr uint16_t NUVOTON_6683_MON_REGISTER_OFFSET	= 0x100;
+	static constexpr uint16_t NUVOTON_6683_MON_CFG_OFFSET		= 0x1A0;
+	static constexpr uint16_t NUVOTON_6683_MON_VOLTAGE_START	= 0x60;
+	static constexpr uint8_t NUVOTON_6683_FAN_NUMS				= 16;
+	static constexpr uint16_t NUVOTON_6683_FAN_REGS[] = { 0x142, 0x140, 0x144, 0x146, 0x148, 0x14A, 0x14C, 0x14E, 0x150, 0x152, 0x154, 0x156, 0x158, 0x15A, 0x15C, 0x15E };
+	static constexpr uint8_t NUVOTON_6683_VOLTAGE_NUMS			= 23;
+	static uint16_t NUVOTON_6683_VOLTAGE_REGS[NUVOTON_6683_VOLTAGE_NUMS] = { 0 };
```
Comment (vit9696) [inline Sensors/SMCSuperIO/NuvotonDevice.hpp]: I do not like this being a global variable, in my opinion, it should really be part of the NuvotonDevice. It should also not be in upper case as it is not a constant.

**Comment by vit9696** [comment]
Looks nice, approved the CI and left a comment on the code. Thx!

File: Sensors/SMCSuperIO/NuvotonDevice.hpp (hunk 1 above)
Comment (chencaidy) [reply Sensors/SMCSuperIO/NuvotonDevice.hpp]: Fixed, Thx for comment!

**Pull request merged**